src/broker/state_machine.c

```diff
@@ -1336,6 +1336,32 @@ static void overlay_monitor_cb (struct overlay *overlay,
     }
 }
 
+static void state_machine_post_cb (flux_t *h,
+                                   flux_msg_handler_t *mh,
+                                   const flux_msg_t *msg,
+                                   void *arg)
+{
+    struct state_machine *s = arg;
+    const char *event;
+    int error = 0;
+
+    if (flux_request_unpack (msg,
+                             NULL,
+                             "{s:s s?b}",
+                             "event", &event,
+                             "error", &error) < 0) {
+        flux_log_error (h, "state-machine.post");
+        return;
+    }
+    if (!flux_msg_is_noresponse (msg)) {
+        (void)flux_respond_error (h, msg, EPROTO, NULL);
+        return;
+    }
+    state_machine_post (s, event);
```

Comment:
Should `event` be validated here? Since it is internally only I'm guessing not necessary, but thought I'd ask.